### ISSUE TITLE
[fuzz-test] Arithmetic overflow occurs while using API Buffer::empty()

### ISSUE THREAD

**Issue by HeeillWang** [ratatui-org/ratatui#517]
## Description

I executed fuzzing to test `tui-0.19.0` APIs. This reproduced on `ratatui` as well.

~~~
Thread '<unnamed>' panicked at 'attempt to multiply with overflow',  tui-0.19.0/src/layout.rs:398
~~~

~~~
    pub fn area(self) -> u16 {
        self.width * self.height //overflow!
    }
~~~

## To Reproduce

~~~
let mut fuzz_arg8:u16 = 65280;
let mut fuzz_arg9:u16 = 65535;
let mut fuzz_arg10:u16 = 65535;
let mut fuzz_arg11:u16 = 65535;
Buffer::empty(Rect { x:fuzz_arg11, y:fuzz_arg9, width:fuzz_arg10, height:fuzz_arg8 });

~~~

## Expected behavior

Adding assert statement or modifying to use checked operation would resolve this issue.

**Comment by a-kenji** [comment]
I think this is incorrect: 
> Arithmetic overflow occurs

No overflow occurred here:
> Thread '<unnamed>' panicked at 'attempt to multiply with overflow'

The thread panicked before an overflow occurred, but please feel free to correct me.

> Adding assert statement

What would that change? Wouldn't an assert just also throw a panic before a possible overflow might occur, basically performing the **same exact** check twice?

> modifying to use checked operation

While I don't think any terminal can get this big and actually function, maybe also providing a checked version of this function does make some sense: `area_checked`.

**Comment by joshka** [comment]
Area should possibly just be u32 or perhaps usize instead to allow for larger dimensions. Note: I haven't checked the history of this decision to understand if there is a good rationale for this. It seems intentional that we avoid allocating more than u16::MAX for any Rect as noted by the ::new() function.

**Comment by HeeillWang** [comment]
Adding assert still meaningful, as arith overflow does not panic by rustc on release build.

Plus, in this particular case, modifying field of `Rect` to `private` or `pub(crate)` can resolve this issue as well. i.e. force external users to use `Rect::new()` instead. Though I'm not very sure if there is another overflow path to `area()` or not...

**Comment by joshka** [comment]
> arith overflow does not panic by rustc on release build.

TIL (thanks).

> Adding assert still meaningful

By definition, u16 * u16 can't overflow u32 / usize. Regardless, the sort of use case where area() is called from is more likely to be for 0 checks rather than > u16::MAX checks. I think a better stopgap solution would be to use saturating ops rather than panicking.

> modifying field of `Rect` to `private` or `pub(crate)` can resolve this issue as well. i.e. force external users to use `Rect::new()` instead

There's a lot of reasonable uses in apps to want to do e.g.:
```rust
let sub_area = Rect { x: area.x + 1, width: area.x - 1, ..area };
```

**Comment by HeeillWang** [comment]
Using saturating ops seems best fit for this case. Thanks.

I'll come up with pr for this issue and #520

**Comment by HeeillWang** [comment]
This has been fixed on #523 , no other issues has been found.
